client/network/src/gossip.rs

```diff
@@ -67,6 +67,9 @@ mod tests;
 pub struct QueuedSender<M> {
 	/// Shared between the front and the back task.
 	shared: Arc<Shared<M>>,
+	// TODO: Can we get around this Mutex? E.g. by taking `&mut` for `lock_queue` and
+	// `queue_or_discard`.
+	notify_background_future: Mutex<Sender<()>>,
```

Comment:
@tomaka I would like to remove the need for this `Mutex`, thus requiring `QueuedSender::lock_queue` and `QueuedSender::queue_or_discard` to take a `&mut self`. Is the current `&self` a design requirement?